pkg/pma/pma.c

```diff
@@ -0,0 +1,860 @@
+#include "pma.h"
+
+#include <assert.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include "page.h"
+#include "sigsegv.h"
+#include "util.h"
+#include "wal.h"
+
+#define PMA_DEBUG
+
+//==============================================================================
+// CONSTANTS
+
+/// Number of bits in a byte.
+static const size_t kBitsPerByte = 8;
+
+/// Number of bits required to track the status of a page. Must be a power of 2
+/// in order to support rounding operations via round_up() and round_down().
+static const size_t kBitsPerPage = 2;
```

Comment:
Maybe at least document that this tracks page_status_t here. i.e. "kBitsPerPage is log2(nmemb(page_status_t))" or in other language. Otherwise, it's unclear the exact point of this constant until you look at its usage.

Reply:
Good point. Done.